Userland/man.cpp

```diff
@@ -45,43 +46,39 @@ int main(int argc, char* argv[])
 
     unveil(nullptr, nullptr);
 
-    String name;
-    String section;
+    const char* name_str = nullptr;
+    int section = -1;
 
-    if (argc < 2 || argc > 3) {
-        fprintf(stderr, "Usage:\t%s <name>\n\t%s <section <name>\n", argv[0], argv[0]);
-        exit(1);
-    }
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(section, "section of the man page", "section", Core::ArgsParser::Required::No);
+    args_parser.add_positional_argument(name_str, "name of the man page", "name");
 
-    if (argc == 2) {
-        name = argv[1];
-    } else {
-        section = argv[1];
-        name = argv[2];
-    }
+    args_parser.parse(argc, argv);
+
+    String name = name_str;
```

Comment:
Do we need it as a `String` below for something?

The whole point of `Core::ArgsParser` using `const char*`s and not `String`s was that you can just reference your argv and avoid copying the string out to the heap. Sometimes you really want it as a `String`, so you convert it like you did; but most of the times you actually don't need that.